### PR TITLE
Upgrade: Multi support for fault-proofs New OP-Stack contracts v3.x.x

### PR DESCRIPTION
**Pull request by nidz-the-fact**
**Upgrade: Multi support for fault-proofs New OP-Stack contracts v3.x.x using "DisputeGameFactory"**
cc.https://docs.optimism.io/stack/fault-proofs/explainer

- **Customizations in this request**
1. Update using viem@2.21.37
2. Add the disputeGameFactory address to the L2 (rollup) chain.

#

- **You can test it.**
1. On local by editing the file .env.test.baseSepolia or .env.test.optimismSepolia According to the name of the network you want to test, set it to .env.
2. Tested on https://test-oni-bridge.pages.dev/ baseSepolia network.

_Thank 🫡🔴💪_

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -17,6 +17,8 @@ dist-ssr
 
 # but allow .env.example
 !.env.example
+!.env.test.baseSepolia
+!.env.test.optimismSepolia
```
Comment (mrmemes-eth) [inline .gitignore]: I don't think it makes sense to have these files included in the repository. It's helpful for testing this PR, but not broadly applicable. Let's remove both of them from the PR and revert this change to `.gitignore`.

File: .env.test.optimismSepolia [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+VITE_L2_CHAIN_NAME=OP Sepolia 
+# separate multiple values with commas; no spaces. single value with no commas is fine.
+VITE_L2_RPC_URLS=https://sepolia.optimism.io
+# URL of the block explorer for the "L2" (rollup) chain
+VITE_L2_EXPLORER_URL=https://optimism-sepolia.blockscout.com
+# The address of the output oracle contract on the "L2" (rollup) chain
+VITE_L2_OUTPUT_ORACLE_ADDRESS=0x90E9c4f8a994a250F6aEfd61CAFb4F2e895D458F
+# The address of the 'disputeGameFactory' contract on the "L2" (rollup) 
+## ++ for Fault proofs - cc.https://docs.optimism.io/stack/fault-proofs/explainer
+VITE_DISPUTE_GAME_FACTORY_ADDRESS=0x05F9613aDB30026FFd634f38e5C4dFd30a197Fa1
```
Comment (mrmemes-eth) [inline .env.test.optimismSepolia]: Can we add a placeholder for this to `.env.example`? That's the canonical reference for what environment variables should be included.

File: src/config.ts [hunk 3]
```diff
@@ -62,6 +62,11 @@ export const rollupChain = defineChain({
   },
   contracts: {
     ...chainConfig.contracts,
+    disputeGameFactory: {
+      [parentChain.id]: {
+        address: import.meta.env.VITE_DISPUTE_GAME_FACTORY_ADDRESS,
+      },
+    },
```
Comment (mrmemes-eth) [inline src/config.ts:69]: Have you tested whether this works for chains that are not using fault proofs? Not sure if we need to make this configuration optional or not.

**Review by mrmemes-eth — COMMENTED** [review]
Thank you very much for the contribution! I have a couple of comments/questions following, but I'm happy to fast track this if we can get them answered. 🍻

File: .gitignore (hunk 1 above)
Comment (nidz-the-fact) [reply .gitignore]: Done ✅

File: .env.test.optimismSepolia (hunk 2 above)
Comment (nidz-the-fact) [reply .env.test.optimismSepolia]: Done ✅

File: src/config.ts (hunk 3 above)
Comment (nidz-the-fact) [reply src/config.ts:69]: Yes, I tested it on a chain that doesn't use **multi** proofs. There is no need to define these things. `DisputeGameFactory` Or you can define it as well.

- [x] Therefore, the address disputeGameFactory set it together is better.

**The topic I propose is mutil proof, you can test it again yourself.**
Topic: https://github.com/chaindrop-raas/oni-bridge/pull/24#issue-2623216370
cc: https://docs.optimism.io/stack/fault-proofs/explainer

**Comment by nidz-the-fact** [comment]
> Thank you very much for the contribution! I have a couple of comments/questions following, but I'm happy to fast track this if we can get them answered. 🍻

**Sure, thank you.** ☝

**Comment by mrmemes-eth** [comment]
Merged! Thanks for the contribution @nidz-the-fact!

**Comment by mrmemes-eth** [comment]
Just confirming, but I was able to use your changes to prove a withdrawal on a chain that did not have fault proofs enabled just now. Thanks again :)
